File: src/test/java/com/ragin/bdd/cucumbertests/database/UserRepository.java
package com.ragin.bdd.cucumbertests.database;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<UserEntity, Long> {
    Optional<UserEntity> findFirstByUserId(String userId);
}

File: src/test/kotlin/com/ragin/bdd/cucumbertests/database/UserRepository.kt
package com.ragin.bdd.cucumbertests.database

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<UserEntity, Long> {
    fun findFirstByUserId(userId: String): Optional<UserEntity>
}
